### PR TITLE
[main] Issue 5383: ContentEncodingContext Builder and passing ContentEncodingContext instance from WebServer to Http1Connection.

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
Added `ContentEncodingSupport.Builder` and removed static content.
`ContentEncodingContext` in `Http1Connection` is taken from `ctx.serverContext()` instead of creating a new instance.

### CODE REVIEW

File: nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java [hunk 1]
```diff
@@ -88,4 +97,109 @@ static ContentEncodingContext create() {
...
+         * <p>
+         * Configuration:<ul>
+         *     <li><b>disable: true</b>  - to disable content encoding support</li>
+         * </ul>
+         *
+         * @param config configuration to use
+         * @return updated builder instance
+         */
+        public Builder config(Config config) {
+            config.get("disable").asBoolean().ifPresent(this::disableFromConfig);
```
Comment (tomas-langer) [inline nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java:134]: The config key should be `enabled`.
Also as we do not really want to handle this this way, the property should be `use-service-loader` or similar (this should already be used somewhere in Helidon).

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java [hunk 2]
```diff
@@ -88,4 +97,109 @@ static ContentEncodingContext create() {
...
+            return this;
+        }
+
+        /**
+         * Disable content encoding support.
+         *
+         * @return updated builder
+         */
+        public Builder disable() {
```
[the remark this answers is not in the document]
Comment (Tomas-Kraus) [reply nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java]: Changed to `public Builder discoverServices(boolean discoverServices)`

File: nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java (hunk 1 above)
Comment (Tomas-Kraus) [reply nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java]: Changed to `discover-services` after some searching for existing implementations of this feature.

File: nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java [hunk 3]
```diff
@@ -88,4 +97,109 @@ static ContentEncodingContext create() {
...
+     */
+    class Builder implements io.helidon.common.Builder<Builder, ContentEncodingContext> {
+
+        private static final String IDENTITY_ENCODING = "identity";
+
+        private final HelidonServiceLoader.Builder<ContentEncodingProvider> encodingProviders
+                = HelidonServiceLoader.builder(ServiceLoader.load(ContentEncodingProvider.class));
+
+        // Force disable of encoders/decoders. Will build empty Maps when false.
+        private boolean enabled = true;
```
[the remark this answers is not in the document]
Comment (Tomas-Kraus) [reply nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java]: Variable removed from the code. Builder method calls `this.encodingProviders.useSystemServiceLoader(discoverServices)` directly.

File: nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java [hunk 4]
```diff
@@ -88,4 +96,97 @@ static ContentEncodingContext create() {
...
+         * @param config configuration to use
+         * @return updated builder instance
+         */
+        public Builder config(Config config) {
+            config.get("discover-services").asBoolean().ifPresent(this::discoverServices);
+            return this;
+        }
+
+        /**
+         * Disable content encoding support.
```
Comment (tomas-langer) [inline nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java:135]: This is not a correct javadoc. It disables discovery from service loader only.

File: nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/WebServer.java [hunk 5]
```diff
@@ -233,9 +233,13 @@ public Builder config(Config config) {
                             connConfig.get("tcp-no-delay").asBoolean().ifPresent(socketOptionsBuilder::tcpNoDelay);
                         });
                     });
+            // Configure content encoding
+            config.get("content-encoding")
```
Comment (tomas-langer) [inline nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/WebServer.java:237]: Use `Config.get("content-encoding").as(ContentEncodingContext::create).ifPresent(this::contentEncodingContext);`

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/WebServer.java (hunk 5 above)
Comment (Tomas-Kraus) [reply nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/WebServer.java:237]: done

File: nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java (hunk 4 above)
Comment (Tomas-Kraus) [reply nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java]: fixed

**Review by tomas-langer — APPROVED** [review]

**Pull request merged**